Src/BootCamp.Chapter/Player.cs

```diff
@@ -73,34 +77,116 @@ public Item[] GetItems(string name)
         // When a slot is equiped, it contributes to total defense
         // and total attack.
         // Implement equiping logic and total defense/attack calculation.
+        private float EquipCapacity { get { return baseCarryWeight + 10 * Strength; } }
         public void Equip(Headpiece head)
         {
-
+            var weightIfEquipped = WeightIfAdded(head) ;
+            if (_equipment.Head != null)
```

Comment:
this check is being repeated, probably could move it into the `WeightIfAdded` method

Reply:
You could even make the method return a bool to really slim down all the equip methods! Then all the logic is in one place, less prone to copy/paste mistakes etc

Reply:
I am having a problem with this part, since it's all checking for different parts, I basically would need a switch case for types

Reply:
if you change the method signature to something like `Item currentItem, Item newItem` then you can check it without any case.

```
if (CanEquipItem(_equipment.Head, head))
    _equipment.Head = head;
```